database/migrations/default/1739497581109_ems_trigger_updates/up.sql

```diff
@@ -0,0 +1,72 @@
+-- delete ems_incidents with the same pcr_key, keeping the first of any dupes
+with numbered_rows as (
+    select
+        id,
+        pcr_key,
+        row_number() over (partition by pcr_key order by id) as row_num
+    from ems__incidents
+)
+
+delete from ems__incidents
+where id in (
+    select id
+    from numbered_rows
+    where row_num > 1
+);
+
+-- add unique constraint
+alter table ems__incidents add constraint unique_pcr_key unique (pcr_key);
+
+-- use BEFORE trigger instead of AFTER
+drop trigger if exists ems_incidents_trigger_insert on ems__incidents;
+drop trigger if exists ems_incidents_trigger_update on ems__incidents;
+
+create trigger ems_incidents_trigger_insert
+before insert
+on ems__incidents
+for each row
+execute function public.ems_incidents_trigger();
+
+create trigger ems_incidents_trigger_update before update on
+ems__incidents for each row when (
+    old.latitude is distinct from new.latitude
+    or old.longitude is distinct from new.longitude
+    or old.apd_incident_numbers is distinct from new.apd_incident_numbers
+    or old.mvc_form_extrication_datetime is distinct from new.mvc_form_extrication_datetime
+) execute function ems_incidents_trigger();
+
+create or replace function public.ems_incidents_trigger()
```

Comment:
the main changes here are:
* setting `geometry` here instead of in the ETL
* using a subquery to assign `in_austin_full_purpose`, which is the same pattern we use for crashes and is faster than the previous use of `union`.